src/controllers/user.js

```diff
@@ -83,4 +84,42 @@ const logout = (req, res) => {
   }
 };
 
-module.exports = { login, register, logout };
+const getUserProfile = async (req, res, next) => {
+  const { username } = req.params;
+  try {
+    const user = await User.findOne({ username });
+    if (!user) return res.status(404).json({ message: 'User not found' });
+
+    const info = {
+      username: user.username,
+      email: user.email,
+      name: user.name,
+      role: user.role,
+      type: user.type,
+    };
+
+    const donations = [];
```

Comment:
new line after const here before the for loop

Reply:
done